### ISSUE TITLE
Convert Base Project Example from Angular to Vanilla

### ISSUE THREAD

**Issue by MatthewVita** [RupertJS/rupert-grunt#3]
David,

npm init has following deps:
```
  "dependencies": {
    "rupert": "^0.2.6",
    "rupert-config-angular": "^0.0.5",
    "rupert-config-bootstrap": "^0.0.2"
  },
```
...should we treat the base project as vanilla js?

**Comment by MatthewVita** [comment]
https://github.com/RupertJS/rupert-grunt/blob/master/dist/plain.tgz#L72L76

**Comment by DavidSouther** [comment]
I'm trying to remember why I'd left it this way... but yeah, that should probably only be `rupert`.

**Comment by MatthewVita** [comment]
The sample app

**Comment by DavidSouther** [comment]
Oh, right - the init app was supposed to be completely contained (eg after npm install, npm test should work). The supporting files will need updating to not depend on the rupert-configs

**Comment by MatthewVita** [comment]
Right. Should we just make it a vanilla js example app instead? I realize it doesn't demonstrate the full power of Rupert, but we don't want people thinking Rupert is dependent/coupled with Angular.

**Comment by DavidSouther** [comment]
Yes - plain should be vanilla. The whirlwind tour still shows the "full" power.

**Comment by MatthewVita** [comment]
Okay we just have to make sure to preface the whirlwind with "Angular" on the main readme. Do we want to stick with the "title" service and related tests with the vanilla js version? I can get started on that... do we want to work in a global sense or use a DI? It's a been a while since I wrote a pure js frontend

**Comment by DavidSouther** [comment]
Just do it as simply as possible, I think. Vanilla js, document.title =
'value'

**Comment by MatthewVita** [comment]
The dream of the 90s is alive in ~~Portland~~ Rupert

jk, nothing wrong with good ole' vanilla JS :)

**Comment by MatthewVita** [comment]
btw, I'm working on this now. I can tell it'll require a good amount of QA between the two of us

**Comment by MatthewVita** [comment]
Okay, so I'm 30 minutes in and am realizing that we should tinker/test/discuss first before I formally submit a PR because I'm noticing some gotchas. Please run the following to get a basic vanilla JS rupert going:

1 ```mkdir ~/test && cd ~/test```
2 ```curl https://raw.githubusercontent.com/RupertJS/rupert-grunt/master/dist/plain.tgz | tar x```
3 ```npm init```
4 ```nano +12 package.json #delete this line```
5 ```npm install```
6 ```nano .jshintrc```:
```
{
  "browser": true,
  "node": false,
  "camelcase": false,
  "eqeqeq": true,
  "indent": 4,
  "quotmark": "single",
  "newcap": true,
  "trailing": true,
  "undef": false,
  "unused": false,
  "latedef": true,
  "globals": {
    "__dirname": true,
    "module": true,
    "require": true
  }
}
```
7 ```cd node_modules/rupert-grunt/src/client```
8 ```nano +68 Gruntfile.coffee #delete this line + the following 2 lines```
9 ```cd ../../../../src/client```
10 ```nano index.jade```
```
doctype html5
html(lang='en')
  head
    meta(charset='utf-8')
    meta(http-equiv='X-UA-Compatible', content='IE=edge,chrome=1')
    meta(name='viewport', content='width=device-width,initial-scale=1')

    title

    link(rel='stylesheet', href='vendors.css')

    link(rel='stylesheet', href='/all.css')
    link(rel='stylesheet', href='/screen.css', media="screen")
    link(rel='stylesheet', href='/print.css', media="print")

    script(src="vendors.js")

    //- Comment below to disable real-time sockets.
    script(src="socket.io/socket.io.js")

  body
    h1 It works!
    //- Put your main content here
    //- .container //- bootstrap
    //- .container-fluid //- Bootstrap Responsive
    //- ion-nav-view //- Root for an ionic layout

    //- Uncomment below for cordova Mobile support
    //- script(src="cordova.js")

    script(src="templates.js")
    script(src="application.js")
```
11 ```rm -rf ./head-controller.js ./title/```
12 ```nano main.js```
```
var TitleService = {
  setTitle: function() {
    document.title = 'Rupert';
  }
};

TitleService.setTitle();
```
13 ```nano main_test.coffee```
```
describe 'Rupert', ->
  it 'has a good title', ->
    document.title.should.equal 'Rupert'
```
14 ```cd ../../```
15 ```grunt && node app.js```

...and it should all work with a title of Rupert, tests succeeded, and a 'It works!' <h1> tag. However I want to bring up these talking point:

- Obviously putting everything in main.js and main_test.coffee is silly (main.js should just have a "switch statement router" via `window.location.pathname` for initialization in this context)... I want to separate them in a /title dir, but it's not recognizing that :(
- I tried to do a main_test.js, but it wasn't recognized. In fact, I think Rupert's grunt fails if there's < 1 client tests present.
- Ripping out those lines from Rupert Grunt are obviously going to screw up the Angular side of things... how would we go about putting that logic into the rupert-angular-config repo?

**Comment by DavidSouther** [comment]
> it's not recognizing that.

You'll need to add to the `server.json`: `{"stassets": { "scripts": { "types": ["*"] }}}` to have Stassets recognize the new script files (this mechanism allows sorting sorting script files in the compiled output).

>  I think Rupert's grunt fails if there's < 1 client tests present

I believe this is a behavior of grunt-karma. I will look more into that.

> Ripping out those lines from Rupert Grunt are obviously going to screw up the Angular side of things

With the previous, I'll make those configurable in `server.json` and `rupert-config-*`.

**Comment by DavidSouther** [comment]
Could you push this code as a PR against rupert-grunt?

**Comment by MatthewVita** [comment]
question: would this basically be me editing the tgz file directly?

**Comment by DavidSouther** [comment]
Huh? no - edit the files in  `./plain/`.

**Comment by MatthewVita** [comment]
I deleted my last comment a minute after I wrote it. I was tired and wasn't thinking... was basically asking if I should commit the tgz/zip files with the PR.

https://github.com/RupertJS/rupert-grunt/pull/4/files

**Comment by DavidSouther** [comment]
No worries - Concurrent with some other work (https://github.com/RupertJS/rupert/issues/19) I took what you'd done here, merged it with some stuff on rupert#19, and pushed it to #5.
